### PR TITLE
Do not use single character variables.

### PR DESCRIPTION
**Pull request by knappe**

### CODE REVIEW

**Comment by dmccraw** [comment]
👍

**Comment by knappe** [comment]
Justification: Other than loops or other common operations the name of a variable is important.  We ought to strive to name things with an appropriate importance.  I like the rule 
> Use short names in small scope, and long names in large scope.

This gives us flexibility to use things like `i` for iterators while also ensuring we use longer names for long lived objects.  Bonus, it also makes navigating the code easier and (in my opinion) improves readability.

File: best_practices/README.md [hunk 1]
```diff
@@ -12,6 +12,7 @@ General
 * [Exceptions should be exceptional].
 * [Keep the code simple].
 * Don't leave trailing whitespace, multiple empty newlines. Always end a file with a newline; standard unix stuff. (`git stripspace` and `git diff --check` are your weapons for this battle)
+* Don't use single character variables.
```
Comment (glanotte) [inline best_practices/README.md]: I whole-heartedly agree, but what about a little elaboration here with regard to exceptions for loops?

Comment (knappe) [inline best_practices/README.md]: I considered it, but given that these are best practices and not rules I figured we didn't need it.  Rules are less likely to be broken than best practices and so I thought that gave us the flexibility to ignore things like using single character variables in loops.

If this doesn't make sense, I'd be glad to add in something.

Comment (knappe) [inline best_practices/README.md]: @glanotte any strong feelings on this?

Comment (glanotte) [inline best_practices/README.md]: It just feels slightly misleading if we really are ok with loop variables being `i`. The wording in your PR for justification was spot-on IMO. Without clarification, you could always point at that line and say "no, these need to be more verbose."

Comment (knappe) [inline best_practices/README.md]: Fair enough, I'll add more of the justification to the wording.

**Comment by knappe** [comment]
@glanotte Updated! https://github.com/teamsnap/guides/pull/31/commits/47b7366c221889046de6e4abf238932f7375ab77

**Pull request merged**